src/libostree/ostree-repo-libarchive.c

```diff
@@ -50,10 +50,10 @@ propagate_libarchive_error (GError      **error,
 static inline const char *
 path_relative (const char *src)
 {
-  if (src[0] == '.' && src[1] == '/')
+  if (src[0] == '.' && src[1] == '\0')
+    return src + 1;
+  while (src[0] == '.' && src[1] == '/')
     src += 2;
-  else if (src[0] == '.' && src[1] == '\0')
-    src++;
   while (src[0] == '/')
     src++;
```

Comment:
This still won't handle `/./foo` right?  I think the code needs to be more like:

```
while (TRUE)
  {
    if (src[0] == '.' && src[1] == '/')
      {
        src += 2;
        continue;
      }
   if (src[0] == '/')
     {
       src++;
       continue
     }
   break;
}
```

Or something?  There's got to be some LGPL-or-below code we can look at as a reference...okay, some quick `git grep relative` in libarchive led me to [strip_absolute_path](https://github.com/libarchive/libarchive/blob/e61afbd463d1b06fcf0d7c0e37feeb9334954af6/tar/util.c#L398) which is rather terrifying...we could likely drop the Windows code at least.